Use update-initramfs, it is the default in Debian/Ubuntu and it also computes and stores some checksums.
with update-initramfs no need to check for do_bootloader and lilo.
--- a/dkms
+++ b/dkms
@@ -207,23 +207,10 @@ function remake_initrd()
     local exitval="0"
     local mkinitrd='mkinitrd'
 
-    # Support /etc/kernel-img.conf (Debian)
-    if [ -f "/etc/kernel-img.conf" ]; then
-        # we use the first one listed
-        mkinitrd=$(sed -e '/^\s*ramdisk/!d' \
-                       -e 's/^\s*ramdisk\s*=\s*\(.*\)/\1/' \
-                       -e "s/['\"]*//g" \
-                       /etc/kernel-img.conf | \
-                   awk '{print $1}')
-    else
-        if type update-initramfs >/dev/null 2>&1; then
-            mkinitrd='update-initramfs'
-        fi
-    fi
 
-    # Support initramfs distributions (Ubuntu).
-    if [ -x "/usr/sbin/mkinitramfs" ]; then
-        mkinitrd='mkinitramfs'
+    # Support initramfs distributions (Debian/Ubuntu).
+    if [ -x "/usr/sbin/update-initramfs" ]; then
+	    mkinitrd='update-initramfs'
     fi
 
     $mkinitrd --version >/dev/null 2>&1
@@ -262,12 +249,10 @@ function remake_initrd()
 	echo $"Updating initrd"
 	echo $"Making new initrd as $initrd_dir/initrd.img-$1"
 	echo $"(If next boot fails, revert to the .bak initrd image)"
-	if [ -f "/etc/initramfs-tools/update-initramfs.conf" ] && \
-	   ! ( grep ^update_initramfs /etc/initramfs-tools/update-initramfs.conf | grep -qi no ) && \
-	   [ "$mkinitrd" == "update-initramfs" ]; then
+	if [ "$mkinitrd" == "update-initramfs" ]; then
 		invoke_command "$mkinitrd -u" "$mkinitrd" background
 	else
-		echo $"Saving old initrd as $initrd_dir/initrd.img_old-$1"
+		echo $"Saving old initrd as $initrd_dir/initrd.img-$1.bak"
 		# we use the same convention as update-initramfs, so that we just
 		# print the warning once
 		cp -f "$initrd_dir/initrd.img-$1" "$initrd_dir/initrd.img-$1.bak"
@@ -281,11 +266,6 @@ function remake_initrd()
 	exitval="$?"
     fi
 
-    # Rerun lilo if necessary
-    if  [ ! -e /etc/kernel-img.conf ] || ( grep ^do_bootloader /etc/kernel-img.conf |  grep -qi yes  ) && [ -e /etc/lilo.conf ]; then
-	invoke_command "/sbin/lilo" "Updating lilo"
-    fi
-
     return $exitval
 }
 
